spyder/plugins/ipythonconsole/widgets/kernelconnect.py

```diff
@@ -129,25 +131,104 @@ def __init__(self, parent=None):
         rm_layout.addWidget(auth_group)
         self.rm_group.setLayout(rm_layout)
         self.rm_group.setCheckable(True)
-        self.rm_group.setChecked(False)
         self.rm_group.toggled.connect(self.pw_radio.setChecked)
 
         # Ok and Cancel buttons
         self.accept_btns = QDialogButtonBox(
             QDialogButtonBox.Ok | QDialogButtonBox.Cancel,
             Qt.Horizontal, self)
 
+        self.accept_btns.accepted.connect(self.save_connection_settings)
         self.accept_btns.accepted.connect(self.accept)
         self.accept_btns.rejected.connect(self.reject)
 
+        # Save connection settings checkbox
+        self.save_layout = QCheckBox(self)
+        self.save_layout.setText(_("Save connection settings"))
+
+        btns_layout = QHBoxLayout()
+        btns_layout.addWidget(self.save_layout)
+        btns_layout.addWidget(self.accept_btns)
+
         # Dialog layout
         layout = QVBoxLayout(self)
         layout.addWidget(main_label)
         layout.addSpacerItem(QSpacerItem(QSpacerItem(0, 8)))
         layout.addLayout(cf_layout)
         layout.addSpacerItem(QSpacerItem(QSpacerItem(0, 12)))
         layout.addWidget(self.rm_group)
-        layout.addWidget(self.accept_btns)
+        layout.addLayout(btns_layout)
+
+        self.load_connection_settings()
+
+    """Loads the user's previously-saved remote kernel connection settings."""
```

Comment:
This needs to go inside the actual function (i.e. *below* the ``def`` line) and should be in imperative tense ("Load the user's...) like all docstrings, commit messages and comments. Sorry for my silly typo originally.